### PR TITLE
Entity implementation

### PR DESCRIPTION
**Pull request by HHjolany**
## Link to Click Up ticket

[ticket (12.4) Api](https://app.clickup.com/t/1da3d6p)
## Describe this PR
due to large file changes this PR is one of the future series PR
### *What is the problem we're trying to solve*

this pull request for entity implementation 

### *What changes have we introduced*

these changes consist of entities, factories, new user data types and enumerations

#### _Checklist_

- [x] Response entity
- [x] Request entity
- [x] Domain
- [x] DynamoDb Table 

### *Follow up actions after merging PR*

depends to the next PR this branch going to be complete on other functionality such as gateway, use case and ...

### CODE REVIEW

File: AccountsApi/V1/Boundary/AccountBaseModel.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+using AccountsApi.V1.Domain;
+using System;
+
+namespace AccountsApi.V1.Boundary
+{
+    public class AccountBaseModel
```
Comment (LBHSPreston) [inline AccountsApi/V1/Boundary/AccountBaseModel.cs:6]: If the AccountBaseModel is never used directly shouldn't we make this an abstract class so we are clear of the intention here?

File: AccountsApi/V1/Infrastructure/AllowedValuesAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        private readonly Type _type;
+
+        public AllowedValuesAttribute(Type enumType)
+        {
+            _type = enumType;
+        }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            if (value == null)
```
Comment (LBHSPreston) [inline AccountsApi/V1/Infrastructure/AllowedValuesAttribute.cs:18]: If we are checking the member name if it is null.  Should we also check that the value is not an empty string or whitespace?

Comment (HHjolany) [reply AccountsApi/V1/Infrastructure/AllowedValuesAttribute.cs:18]: do you mean the **ValidationContext validationContext**? if yes, is it possible to happen?

File: AccountsApi/V1/Boundary/AccountBaseModel.cs (hunk 1 above)
Comment (HHjolany) [reply AccountsApi/V1/Boundary/AccountBaseModel.cs:6]: Sure.

File: AccountsApi/V1/Infrastructure/AllowedValuesAttribute.cs (hunk 2 above)
Comment (LBHSPreston) [reply AccountsApi/V1/Infrastructure/AllowedValuesAttribute.cs:18]: Yes that's the one.

Comment (HHjolany) [reply AccountsApi/V1/Infrastructure/AllowedValuesAttribute.cs:18]: thanks / done

File: AccountsApi/V1/Boundary/AccountBaseModel.cs (hunk 1 above)
Comment (LBHSPreston) [reply AccountsApi/V1/Boundary/AccountBaseModel.cs]: Did you make the AccountBaseModel class abstract?  I'm not seeing the update.

File: AccountsApi/V1/Infrastructure/AllowedValuesAttribute.cs (hunk 2 above)
Comment (LBHSPreston) [reply AccountsApi/V1/Infrastructure/AllowedValuesAttribute.cs:24]: I'm not seeing the change.  Was it done on a different branch?

File: AccountsApi/V1/Boundary/AccountBaseModel.cs (hunk 1 above)
Comment (HHjolany) [reply AccountsApi/V1/Boundary/AccountBaseModel.cs]: yes i did
[https://github.com/LBHackney-IT/account-api/pull/17/files](https://github.com/LBHackney-IT/account-api/pull/17/files)

**Review by LBHSPreston — APPROVED** [review]

**Pull request merged**